### PR TITLE
French translation for WSO2 IS-5.11

### PR DESCRIPTION
**Pull request by geve82**
## Purpose
French translation Contribution from Smile (Gregory Eve and Hery njato RANDRIAMANAMIHAGA)

## Goals
Add french translation for all apps

## Approach
Enable localization support and put all hardcoded string in Resources.properties for:
- email-otp-authentication-portal
- sms-otp-authentication-portal
- totp-authentication-portal
- x509-certificate-authentication-portal

Added a Resources_fr-Fr.properties for:
- authentication-portal
- email-otp-authentication-portal
- recovery-portal
- sms-otp-authentication-portal
- totp-authentication-portal
- x509-certificate-authentication-portal

Add fr-FR language in modules/i18n/src/translations and translated:
- admin-portal.ts
- common.ts
- console.ts
- dev-portal.ts
- index.ts
- user-portal.ts

Add a quick fix on a hardcoded button "generate password" to be translatable.

Lots of strings remain in English as they are hardcoded for example:
- item per page on tables
- application templates
- authenticators
- etc.
## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
N/A

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
WSO2-IS-5.11-Alpha3
 
## Learning
N/A

### CODE REVIEW

File: modules/i18n/src/translations/en-US/portals/console.ts [hunk 1]
```diff
@@ -26,66 +26,66 @@ export const console: ConsoleNS = {
                     gravatar: {
                         errors: {
                             noAssociation: {
-                                content: "It seems like the selected email is not registered on Gravatar. " +
-                                    "Sign up for a Gravatar account by visiting Gravatar official website or use " +
-                                    "one of the following.",
-                                header: "No matching Gravatar image found!"
+                                content: "Il semble que l'e-mail sélectionné ne soit pas enregistré sur Gravatar. " +
```
Comment (brionmario) [inline modules/i18n/src/translations/en-US/portals/console.ts:29]: This is the `en-US` translation file, hence french translations here is not needed right?

File: modules/i18n/src/translations/fr-FR/portals/common.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    terminateAll: "Tout terminer",
+    terminateSession: "Terminer la session",
+    type: "Type",
+    unpin: "Détacher",
+    unpinned: "Détaché",
+    update: "Mettre à jour",
+    user: "Utilisateur",
+    verify: "Vérifier",
+    view: "Voir"
+};
```
Comment (brionmario) [inline modules/i18n/src/translations/fr-FR/portals/common.ts:134]: Missing new line.

File: modules/i18n/src/translations/si-LK/portals/common.ts [hunk 3]
```diff
@@ -115,6 +115,7 @@ export const common: CommonNS = {
     showLess: "අඩුවෙන් පෙන්වන්න",
     showMore: "වැඩිදුර තොරතුරු",
     showPassword: "මුරපදය පෙන්වන්න",
+    generatePassword: "Generate password",
```
Comment (brionmario) [inline modules/i18n/src/translations/si-LK/portals/common.ts:118]:
```suggestion
    generatePassword: "මුරපදය ජනනය කරන්න",
```

File: modules/i18n/src/translations/pt-BR/portals/common.ts [hunk 4]
```diff
@@ -115,6 +115,7 @@ export const common: CommonNS = {
     showLess: "Mostre menos",
     showMore: "Mostre mais",
     showPassword: "Mostrar senha",
+    generatePassword: "Generate password",
```
Comment (brionmario) [inline modules/i18n/src/translations/pt-BR/portals/common.ts:118]:
```suggestion
    generatePassword: "Gerar senha",
```

File: modules/i18n/src/translations/ta-IN/portals/common.ts [hunk 5]
```diff
@@ -115,6 +115,7 @@ export const common: CommonNS = {
     showLess: "சுருக்கிக் காண்",
     showMore: "மேலுங் காண்",
     showPassword: "கடவுச்சொல்லை காண்",
+    generatePassword: "Generate password",
```
Comment (brionmario) [inline modules/i18n/src/translations/ta-IN/portals/common.ts:118]: @thivi Please validate.

```suggestion
    generatePassword: "கடவுச்சொல்லை உருவாக்கவும்",
```

**Review by brionmario — CHANGES_REQUESTED** [review]

**Comment by brionmario** [comment]
Hi @geve82 & @hr15a,

Thanks a lot for this PR. Added few comments. Would you be able to address them?

File: modules/i18n/src/translations/en-US/portals/console.ts (hunk 1 above)
Comment (geve82) [reply modules/i18n/src/translations/en-US/portals/console.ts:29]: good catch we have inverted fr and en file. I change it.

File: modules/i18n/src/translations/fr-FR/portals/common.ts (hunk 2 above)
Comment (geve82) [reply modules/i18n/src/translations/fr-FR/portals/common.ts]: resolved

**Review by geve82 — COMMENTED** [review]
Hi @brionmario all the changes requested has been done.

**Review by thivi — APPROVED** [review]

**Review by thanujalk — APPROVED** [review]

**Review by DimalChandrasiri — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

**Comment by brionmario** [comment]
> Hi @brionmario all the changes requested has been done.

Approved. Thanks a lot.

**Pull request merged**